ARCHITECTURE.md

```diff
@@ -0,0 +1,103 @@
+
+### Standalone
+![](https://raw.githubusercontent.com/balenalabs/balena-sound/master/docs/images/arch-standalone.png)
+
+Standalone mode is easy to understand. You just pipe ` balena-sound.input` to `balena-sound.output` and that's it. Audio coming in from any plugin will find it's way to the selected output. If this was the only mode, we could simplify the setup and use a single sink. Having the two layers however is important for the next mode which is more complicated.
+
+
+### Multiroom
+![](https://raw.githubusercontent.com/balenalabs/balena-sound/master/docs/images/arch-multiroom.png)
+
+Multiroom feature relies on `snapcast` to broadcast the audio to multiple devices. Snapcast has two binaries working alonside, server and client.
+
+Snapcast server expects audio to be written into a FIFO file, so we create an additional sink (`snapcast` sink) that routes audio from `balena-sound.input` into said FIFO file. The server will then read the file and use TCP packets to broadcast audio to all clients that are connected to it, wether they run in the same device or others. Note that when writting into the FIFO file the audio is "exiting" the `audio` block and no longer under PulseAudio's control.
+
+Snapcast client receives the audio from the server and sends it back into the `audio` block, in particular to `balena-sound.output` sink which will in turn send the audio to whatever output was selected by the user.
+
+This setup allows us to decouple the multiroom feature from the `audio` block while retaining it's advantages.
+
+## Plugins
+
+As described above, plugins are the services generating the audio to be streamed/played. Plugins are responsible for sending the audio into the `audio` block, particularily into `balena-sound.input` sink. There are two alternatives for how this can be acomplished. A detailed explanation can be found [here](https://github.com/balenablocks/audio#usage), in our case:
+
+**PulseAudio backend**
+Most audio applications support using PulseAudio as an audio backend. This means the application was coded to allow sending audio directly to PulseAudio (and hence the `audio` block). This is usually configurable via a CLI option flag or configuration files. You should check your application's documentation and figure out if this is the case.
+
+If the application supports PulseAudio backend, the only configuration you need is to specify where the PulseAudio server can be located. This can be done by setting the `PULSE_SERVER` environment variable, we recommend doing it in the `Dockerfile`:
+
+```
+ENV PULSE_SERVER=tcp:localhost:4317
+```
+
+**ALSA bridge**
+If your application does not have built-in PulseAudio support you can create a bridge to it by using ALSA. This can't be one-lined so we wrote a little script that will do the work for you:
```

Comment:
```suggestion
If your application does not have built-in PulseAudio support, you can create a bridge to it by using ALSA. This can't be added in easily, so we wrote a little script that will do the work for you:
```